### PR TITLE
ARACHNE-2123 - Add setting to include / exclude some file types

### PR DESCRIPTION
**Pull request by mpozhidaeva**

### CODE REVIEW

File: execution-engine-commons/src/main/java/com/odysseusinc/arachne/execution_engine_common/util/CommonFileUtils.java [hunk 1]
```diff
@@ -106,4 +110,23 @@ public static File compressAndSplit(File folder, File zipArchive, Long maximumSi
         }
         return zipDir;
     }
+
+    private static ArrayList<File> filterFiles(Path folderPath, String exclusions) throws IOException {
+
+        List<String> patterns = Lists.newArrayList(split(exclusions, ","));
+        AntPathMatcher antPathMatcher = new AntPathMatcher();
+
```
Comment (wivern) [inline execution-engine-commons/src/main/java/com/odysseusinc/arachne/execution_engine_common/util/CommonFileUtils.java]: It seems that antPathMatcher can be moved to noneMatch method

File: arachne-sys-settings/src/main/java/com/odysseusinc/arachne/system/settings/service/impl/SystemSettingsServiceImpl.java [hunk 2]
```diff
@@ -123,18 +132,36 @@ public boolean isSecuredSetting(SystemSetting systemSetting) {
...
-    private void updateProperty(List<SystemSetting> forUpdate, Map.Entry<Long, String> entry, SystemSetting systemProperty) {
+    private void updateProperty(List<SystemSetting> forUpdate, Map.Entry<Long, String> entry,
+                                SystemSetting systemProperty) throws BindException {
 
         String value = entry.getValue();
         if (value != null) {
             if (isSecuredSetting(systemProperty)) {
                 value = getEncryptedValue(value);
             }
+            if (PATTERN_SETTING_NAME.equals(systemProperty.getName())) {
```
Comment (pavgra) [inline arachne-sys-settings/src/main/java/com/odysseusinc/arachne/system/settings/service/impl/SystemSettingsServiceImpl.java]: So are you going to add `if` for each new field, which requires such check? Why haven't you added new entry into SystemSettingType?

Comment (mpozhidaeva) [reply arachne-sys-settings/src/main/java/com/odysseusinc/arachne/system/settings/service/impl/SystemSettingsServiceImpl.java]: if the second field with patterns appears - I will add a new type

File: execution-engine-commons/src/main/java/com/odysseusinc/arachne/execution_engine_common/api/v1/dto/AnalysisRequestDTO.java [hunk 3]
```diff
@@ -50,6 +50,8 @@
     @NotNull
     private Date requested;
 
+    private String exclusions = "";
```
Comment (pavgra) [inline execution-engine-commons/src/main/java/com/odysseusinc/arachne/execution_engine_common/api/v1/dto/AnalysisRequestDTO.java]: resultExclusions

**Pull request merged**